src/pages/[guild]/members.tsx

```diff
@@ -0,0 +1,227 @@
+import { Checkbox, HStack, Text } from "@chakra-ui/react"
+import {
+  createColumnHelper,
+  getCoreRowModel,
+  useReactTable,
+} from "@tanstack/react-table"
+import GuildTabs from "components/[guild]/Tabs/GuildTabs"
+import { ThemeProvider, useThemeContext } from "components/[guild]/ThemeContext"
+import CRMTable from "components/[guild]/crm/CRMTable"
+import ExportMembers from "components/[guild]/crm/ExportMembers"
+import FilterByRoles from "components/[guild]/crm/FilterByRoles"
+import Identities from "components/[guild]/crm/Identities"
+import IdentitiesSearch from "components/[guild]/crm/IdentitiesSearch"
+import OrderByColumn from "components/[guild]/crm/OrderByColumn"
+import RoleTags from "components/[guild]/crm/RoleTags"
+import {
+  buildQueryStringFromState,
+  parseFiltersFromQuery,
+  parseSortingFromQuery,
+} from "components/[guild]/crm/transformTableStateToAndFromQuery"
+import useMembers, { Member } from "components/[guild]/crm/useMembers"
+import useGuild from "components/[guild]/hooks/useGuild"
+import GuildLogo from "components/common/GuildLogo"
+import Layout from "components/common/Layout"
+import Head from "next/head"
+import { useRouter } from "next/router"
+import ErrorPage from "pages/_error"
+import { useEffect, useMemo, useState } from "react"
+import { Visibility } from "types"
+
+const columnHelper = createColumnHelper<Member>()
+const getRowId = (row: Member) => `user_${row.userId}`
+
+const GuildPage = (): JSX.Element => {
+  const { textColor, localThemeColor, localBackgroundImage } = useThemeContext()
+  const { name, roles, imageUrl } = useGuild()
+  const hasHiddenRoles = roles?.some((role) => role.visibility === Visibility.HIDDEN)
+
+  const router = useRouter()
+  const [columnFilters, setColumnFilters] = useState(() =>
+    parseFiltersFromQuery(router.query)
+  )
+  const [sorting, setSorting] = useState(() => parseSortingFromQuery(router.query))
+  const [rowSelection, setRowSelection] = useState({})
+
+  const queryString = useMemo(
+    () => buildQueryStringFromState(columnFilters, sorting),
+    [columnFilters, sorting]
+  )
+
+  useEffect(() => {
+    window.history.pushState("", "", `?${queryString}`)
```

Comment:
Can't we use next router here? I think the browser navigation doesn't work properly this way. (at least if I set some filters on the members page, then visit another page, e.g. "Events", then click on the back button in the browser, the app won't navigate back to the "Members" page)